pkg/router/route_group.go

```diff
@@ -402,6 +407,31 @@ func (rg *RouteGroup) sendKeepAlive() error {
 	return nil
 }
 
+func (rg *RouteGroup) sendHandshake(encrypt bool) error {
+	rg.mu.Lock()
+
+	if len(rg.tps) == 0 || len(rg.fwd) == 0 {
+		rg.mu.Unlock()
+		// if no transports, no rules, then no keepalive
+		return nil
+	}
+
+	rule := rg.fwd[0]
```

Comment:
Could you please handle all rules from `rg.fwd` and transports from `rg.tps` instead of just the 1st ones?

Reply:
Done in the following commits.